File: Data Structures & Algorithms/palindrome-linked-list/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func isPalindrome(head *ListNode) bool {
	i := 0
	j := 0
	dictPal := make(map[int]int)
	for head != nil {
		dictPal[i] = head.Val
		fmt.Println(head.Val)
		i++
		head = head.Next
	}
	
	i = i - 1
	for j < i {
		fmt.Println(j)
		fmt.Println(i)
		if dictPal[j] != dictPal[i] {
			return false
		}
		j++
		i--
	}
	return true
}

File: Data Structures & Algorithms/reorder-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        mynodes = {}
        i = 0
        track = head
        while track != None:
            mynodes[i] = track
            track = track.next
            i+=1
        
        left = 0
        right = i-1
        while left < right:
            print(left)
            mynodes[left].next = mynodes[right]
            left+=1
            if left == right:
                break
            else:
                mynodes[right].next = mynodes[left]
                right -=1
        
        mynodes[left].next = None